storage/innobase/trx/trx0purge.cc

```diff
@@ -163,6 +85,13 @@ purge_graph_build()
 	return(fork);
 }
 
+ATTRIBUTE_NOINLINE /* this is a rather complex operation */
+void purge_sys_t::enqueue(trx_rseg_t &rseg)
```

Comment:
Given that all calls after server startup are unconditional from this same compilation unit from `trx_t::write_serialisation_history()`, it could make sense to omit the `ATTRIBUTE_NOINLINE`. I would suggest that you benchmark it once more.